src/goodbyedpi.c

```diff
@@ -189,6 +185,19 @@ static char* dumb_memmem(const char* haystack, int hlen, const char* needle, int
     return NULL;
 }
 
+unsigned short int atousi(const char *str, const char *msg){
+    long unsigned int res = strtoul(str, NULL, 10u);
+    enum{
+        limitValue=0xFFFFu
+    };
+    if(res > limitValue){
```

Comment:
1 Modern compilers are smart, we shouldn't do their work instead of them.
2 If you wanna inline, you should tell compiler about it, not use macrodef. In my experience macrodefs is the way to "happy debugging".